src/main/kotlin/com/expediagroup/sdk/core/configuration/collector/ConfigurationCollector.kt

```diff
@@ -43,8 +50,12 @@ internal class ConfigurationCollector private constructor(providers: Configurati
         fun create(vararg providers: ConfigurationProvider): ConfigurationCollector = create(ConfigurationProviderQueue.from(providers.asList()))
     }
 
-    override val key: String by lazy { providers.firstOf { it.key } }
-    override val secret: String by lazy { providers.firstOf { it.secret } }
-    override val endpoint: String by lazy { providers.firstOf { it.endpoint } }
-    override val authEndpoint: String by lazy { providers.firstOf { it.authEndpoint } }
+    val key: String by lazy { providers.firstOf { provider -> provider.key.also { logConfigurationRetrieval(provider, KEY) } } }
+    val secret: String by lazy { providers.firstOf { provider -> provider.secret.also { logConfigurationRetrieval(provider, SECRET) } } }
+    val endpoint: String by lazy { providers.firstOf { provider -> provider.endpoint.also { logConfigurationRetrieval(provider, ENDPOINT) } } }
+    val authEndpoint: String by lazy { providers.firstOf { provider -> provider.authEndpoint.also { logConfigurationRetrieval(provider, AUTH_ENDPOINT) } } }
+
+    private fun logConfigurationRetrieval(provider: ConfigurationProvider, configurationName: String) {
```

Comment:
We may define this as an extension function inside this class (with a short name, e.g., `log`).